### ISSUE TITLE
Microservice helm charts use deprecated k8s API

### ISSUE THREAD

**Issue by djones6** [ibm-cloud-architecture/refarch-kc#77]
The microservices will not install on newer levels of Kubernetes (1.16.x) as [some deprecated APIs have been removed](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) (eg. deployments that specify extensions/v1beta1 will no longer work, and should be updated to apps/v1).

Symptom:
```
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1"
```

Environment that this was observed:
```
$kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:18:29Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by djones6** [comment]
In a similar vein, it might be worth deciding what the minimum level of Kubernetes is that we're interested in supporting.  `apps/v1` for example has been around since 1.9, which is now fairly old.  I did come across some places in charts where an attempt is made to support older levels.  If that's no longer relevant then the charts could be simplified.

**Comment by osowski** [comment]
Good catch. I don't think there's any reason why we would need to support anything-k8s-wise that is less than what is included in OCP 3.11 - [which would be k8s 1.11](https://docs.openshift.com/container-platform/3.11/release_notes/ocp_3_11_release_notes.html).

We can run through and make a check of all the Charts types, as they were originally generated in silos across the team and using different CLI/scaffolding tools.

**Comment by kylegc** [comment]
I was blocked by this so worked on it myself. There were a few updated that I needed to get it running locally on Docker for Desktop
```
$kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:18:29Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

1. Convert Deployment resources to use `apiVersion: apps/v1`
1. Add `spec.selector` to Deployment resources
1. Convert Istio resources `apiVersion: networking.k8s.io/v1beta1`

I'm working on creating and linking the PRs in all the repos now.

**Comment by kylegc** [comment]
PRs:
https://github.com/ibm-cloud-architecture/refarch-kc-container-ms/pull/44
https://github.com/ibm-cloud-architecture/refarch-kc-order-ms/pull/65
https://github.com/ibm-cloud-architecture/refarch-kc-ui/pull/66
https://github.com/ibm-cloud-architecture/refarch-kc-ms/pull/46
https://github.com/ibm-cloud-architecture/refarch-reefer-ml/pull/61

**Comment by jesusmah** [comment]
@kylegc just reviewed and accepted your PRs. Thanks a lot for your work here. I's very much appreciated given the short squad we are. Thanks

**Comment by kylegc** [comment]
No problem, thank you!
